tests/test_tlsfuzzer_expect.py

```diff
@@ -1285,52 +1285,140 @@ def test_process_with_tls13_unallowed_extension(self):
         with self.assertRaises(AssertionError):
             exp.process(state, server_hello)
 
-    def test_process_with_tls_1_3_no_downgrade_protection(self):
-        exp = ExpectServerHello(version=(3, 4), server_max_protocol=(3, 4))
+    def test_process_with_tls_1_3_in_legacy_version(self):
+        exp = ExpectServerHello()
 
         state = ConnectionState()
         client_hello = ClientHello()
         ciph = CipherSuite.TLS_EMPTY_RENEGOTIATION_INFO_SCSV
         client_hello.cipher_suites = [4, ciph]
         state.handshake_messages.append(client_hello)
         state.msg_sock = mock.MagicMock()
+        ext = []
+        ext.append(SrvSupportedVersionsExtension().create((3, 4)))
 
         msg = ServerHello().create(version=(3, 4),
                                    random=bytearray(32),
                                    session_id=bytearray(0),
                                    cipher_suite=4,
-                                   extensions=None)
+                                   extensions=ext)
+
+        self.assertTrue(exp.is_match(msg))
+
+        with self.assertRaises(ValueError) as e:
+            exp.process(state, msg)
+
+        self.assertIn("invalid version in legacy_version", str(e.exception))
+
+    def test_process_with_tls_1_3_no_downgrade_protection(self):
+        # use default extension handlers
+        exp = ExpectServerHello(version=(3, 3), server_max_protocol=(3, 4))
+
+        state = ConnectionState()
+        client_hello = ClientHello()
+        client_hello.extensions = []
+        client_hello.cipher_suites = [CipherSuite.TLS_AES_128_GCM_SHA256]
+        ext = SupportedGroupsExtension().create([GroupName.secp256r1])
+        client_hello.extensions.append(ext)
+        c_ks = key_share_gen(GroupName.secp256r1)
+        ext = ClientKeyShareExtension().create([c_ks])
+        client_hello.extensions.append(ext)
+        ext = SupportedVersionsExtension().create([(3, 3), (3, 4)])
+        client_hello.extensions.append(ext)
+        state.handshake_messages.append(client_hello)
+        state.msg_sock = mock.MagicMock()
+        ext = []
+        ext.append(SrvSupportedVersionsExtension().create((3, 4)))
+
+        msg = ServerHello().create(version=(3, 3),
+                                   random=bytearray(32),
+                                   session_id=bytearray(0),
+                                   cipher_suite=
+                                   CipherSuite.TLS_AES_128_GCM_SHA256,
+                                   extensions=ext)
 
         self.assertTrue(exp.is_match(msg))
 
         exp.process(state, msg)
 
-    def test_process_with_tls_1_3_wrong_downgrade_protection(self):
-        exp = ExpectServerHello(extensions={ExtensionType.renegotiation_info:
-                                            None},
-                                version=(3, 4), server_max_protocol=(3, 4))
+    def test_process_with_default_and_tls_1_3_with_1_2_downgrade_sentinel(self):
```

Comment:
name of test is confusing to me

Reply:
changed it to `test_process_with_default_settings_and_tls_1_3_reply_with_1_2_downgrade_sentinel` and extended the comment below, does that make it clearer? if not any other suggestion how to make it clear?